omx/gstomxcamera.c

```diff
@@ -0,0 +1,1214 @@
+gst_omx_camera_get_property (GObject * object,
+    guint prop_id, GValue * value, GParamSpec * pspec)
+{
+  GstOMXCamera *self = GST_OMX_CAMERA (object);
+  OMX_ERRORTYPE err;
+
+  switch (prop_id) {
+    case PROP_INTERFACE:
+      g_value_set_enum (value, self->interface);
+      break;
+    case PROP_CAPT_MODE:
+      g_value_set_enum (value, self->capt_mode);
+      break;
+    case PROP_VIP_MODE:
+      g_value_set_enum (value, self->vip_mode);
+      break;
+    case PROP_SCAN_TYPE:
+      g_value_set_enum (value, self->scan_type);
+      break;
+    case PROP_ALWAYS_COPY:
+      g_value_set_boolean (value, self->always_copy);
+      break;
+    case PROP_NUM_OUT_BUFFERS:
+      g_value_set_uint (value, self->num_buffers);
+      break;
+    case PROP_SKIP_FRAMES:
+    {
+      if (self->comp) {
+        OMX_CONFIG_VFCC_FRAMESKIP_INFO skip_frames;
+
+        err =
+            gst_omx_component_set_config (self->comp,
+            OMX_TI_IndexConfigVFCCFrameSkip, &skip_frames);
+        if (err != OMX_ErrorNone)
+          GST_ERROR_OBJECT (self,
+              "Failed to get capture skip frames: %s (0x%08x)",
+              gst_omx_error_to_string (err), err);
+
+        g_value_set_uint (value, skip_frames.frameSkipMask);
+      } else {
+        g_value_set_uint (value, self->skip_frames);
+      }
+      break;
+    }
+    default:
+      G_OBJECT_WARN_INVALID_PROPERTY_ID (object, prop_id, pspec);
+      break;
+  }
+}
+
+/* Following caps negotiation related functions were taken from the
+ * omx_camera element code */
+
+/* this function is a bit of a last resort */
+static GstCaps *
+gst_omx_camera_fixate (GstBaseSrc * basesrc, GstCaps * caps)
+{
+  GstStructure *structure;
+  gint i;
+
+  GST_DEBUG_OBJECT (basesrc, "fixating caps %" GST_PTR_FORMAT, caps);
+
+  caps = gst_caps_make_writable (caps);
+
+  for (i = 0; i < gst_caps_get_size (caps); ++i) {
+    structure = gst_caps_get_structure (caps, i);
+
+    /* We are fixating to a resonable 320x200 resolution
+       and the maximum framerate resolution for that size */
+    gst_structure_fixate_field_nearest_int (structure, "width", 320);
+    gst_structure_fixate_field_nearest_int (structure, "height", 240);
+    gst_structure_fixate_field_nearest_fraction (structure, "framerate",
+        G_MAXINT, 1);
+    gst_structure_fixate_field (structure, "format");
+  }
+
+  GST_DEBUG_OBJECT (basesrc, "fixated caps %" GST_PTR_FORMAT, caps);
+
+  caps = GST_BASE_SRC_CLASS (parent_class)->fixate (basesrc, caps);
+
+  return caps;
+}
+
+static gboolean
+gst_omx_camera_event (GstBaseSrc * src, GstEvent * event)
+{
+  GST_DEBUG_OBJECT (src, "handle event %" GST_PTR_FORMAT, event);
+
+  switch (GST_EVENT_TYPE (event)) {
+    case GST_EVENT_RECONFIGURE:
+      gst_pad_check_reconfigure (GST_BASE_SRC_PAD (src));
+      break;
+    default:
+      break;
+  }
+  return GST_BASE_SRC_CLASS (parent_class)->event (src, event);
+}
+
+
+static gboolean
+gst_omx_camera_negotiate (GstBaseSrc * basesrc)
```

Comment:
I don't think overriding this method is necessary either. There's no special logic, base class should handle it well.